src/Tools/AI Test Toolkit/src/TestSuite/AITALTestSuiteMgt.Codeunit.al

```diff
@@ -74,17 +74,21 @@ codeunit 149037 "AIT AL Test Suite Mgt"
         TempTestMethodLine: Record "Test Method Line" temporary;
         ALTestSuite: Record "AL Test Suite";
         TestInputsManagement: Codeunit "Test Inputs Management";
+        Iteration: Integer;
     begin
         ALTestSuite := GetOrCreateALTestSuite(AITTestMethodLine);
 
-        TempTestMethodLine."Line Type" := TempTestMethodLine."Line Type"::Codeunit;
-        TempTestMethodLine."Test Codeunit" := AITTestMethodLine."Codeunit ID";
-        TempTestMethodLine."Test Suite" := AITTestMethodLine."AL Test Suite";
-        TempTestMethodLine."Data Input Group Code" := TestInput."Test Input Group Code";
-        TempTestMethodLine."Data Input" := TestInput.Code;
-        TempTestMethodLine.Insert();
+        for Iteration := 0 to AITTestMethodLine.Iterations - 1 do begin
```

Comment:
I would not do it like this.
I would implement the code in the Run method on the suite to mimic like user has invoked this action e.g. 20 times for 20 iterations. This would be the best design because this is the flow that we are implementing.

Reply:
Then we should also add functionality to allow easy access to aggregated results across multiple versions. Otherwise, it's going to be difficult to get an average accuracy number, which is the main purpose of allowing multiple iterations.